### PR TITLE
specs, add test for encode integer as string

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Fix https://github.com/Azure/cadl-ranch/issues/634

Not include case of "query" and "header", because they are always string on wire.

"body property" tests on
1. `safeint` (int64) as required property
2. `uint32` as optional property

# Cadl Ranch Contribution Checklist:

- [x] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [x] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [x] I have written a [mock API](../docs/writing-mock-apis.md)
- [x] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/http/encode/string/main.tsp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    "value": "1"
+  }
+  ```
+  """)
+  @post
+  op encodeAsString(@body body: IntegerAsStringProperty): IntegerAsStringProperty;
+
+  model IntegerAsStringProperty {
+    @encode(string)
+    value: safeint;
```
Comment (archerzz) [inline packages/cadl-ranch-specs/http/encode/string/main.tsp:38]: Shall we add a series of test cases like `integer`, `int8`... etc?

File: packages/cadl-ranch-specs/http/encode/string/main.tsp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  {
+    "value": "1"
+  }
+  ```
+  """)
+  @post
+  op encodeAsString(@body body: IntegerAsStringProperty): IntegerAsStringProperty;
+
+  model IntegerAsStringProperty {
+    @encode(string)
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/encode/string/main.tsp:37]: Current TCGC does not handle the ModelProperty path. `@encode(string) value: safeint;` does not work.

It has to be
```
  @encode(string)
  scalar safeintEncodedAsString extends safeint;

  model IntegerAsStringProperty {
    value: safeintEncodedAsString;
  }
```

image

Need a fix.

File: packages/cadl-ranch-specs/http/encode/string/main.tsp (hunk 1 above)
Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/encode/string/main.tsp:38]: Personally I'd avoid adding type that would be same client type for language.

Let me know what matters in your and I can add them.

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/encode/string/main.tsp:38]: I've added another `int32` (optional property).

**Review by haolingdong-msft — APPROVED** [review]

File: packages/cadl-ranch-specs/http/encode/string/main.tsp [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+@operationGroup
+@route("/property")
+namespace Property {
+  @route("/safeint")
+  @scenario
+  @scenarioDoc("""
+    Test operation with request and response model contains integer property with string encode.
+    Expected request body:
+    ```json
```
Comment (msyyc) [inline packages/cadl-ranch-specs/http/encode/string/main.tsp:21]: One thing to confirm: for now we only support for int, not including float/decimal?

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/encode/string/main.tsp:21]: I am not exactly sure which is the final decision (long discussion and I wasn't in the meeting that reached decision).

I assume Laurent want it simple as int to string https://github.com/Azure/typespec-azure/issues/1179#issuecomment-2233697638
Me too...

**Review by tadelesh — COMMENTED** [review]
shall we change the folder to numeric? the previous tests for encode is named by original type name.

**Review by msyyc — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
> shall we change the folder to numeric? the previous tests for encode is named by original type name.

Make sense. Moved to `encode/numeric` folder (and namespace).

**Review by archerzz — APPROVED** [review]

**Review by tadelesh — APPROVED** [review]

File: packages/cadl-ranch-specs/http/encode/string/main.tsp (hunk 3 above)
Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/encode/string/main.tsp]: From current lint, it appears that other numeric like `float32` etc. also won't trigger a error (`boolean` or `url` do).

**Pull request merged**